src/Microsoft.TemplateEngine.Cli/PostActionProcessors/DotnetBuildPostActionProcessor.cs

```diff
@@ -0,0 +1,141 @@
+    {
+        private static readonly Guid ActionProcessorId = new Guid("210A437B-A78E-4D2F-B562-4DD3E3EA4025");
+
+        public Guid Id => ActionProcessorId;
+
+        public bool Process(IEngineEnvironmentSettings environment, IPostAction actionConfig, ICreationEffects2 creationEffects, ICreationResult templateCreationResult, string outputBasePath)
+        {
+            bool allSucceeded = true;
+            IEnumerable<string> targetFiles = null;
+
+            if (actionConfig.Args.TryGetValue("files", out string specificFilesString))
+            {
+                JToken config = JToken.Parse(specificFilesString);
+
+                if (config.Type == JTokenType.String)
+                {
+                    targetFiles = specificFilesString.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries).SelectMany(x => GetTargetForSource(creationEffects, x));
+                }
+                else if (config is JArray arr)
+                {
+                    List<string> allFiles = new List<string>();
+                    foreach (JToken child in arr)
+                    {
+                        if (child.Type != JTokenType.String)
+                        {
+                            continue;
+                        }
+
+                        allFiles.AddRange(GetTargetForSource(creationEffects, child.ToString()));
+                    }
+
+                    if (allFiles.Count > 0)
+                    {
+                        targetFiles = allFiles;
+                    }
+                }
+            }
+            else
+            {
+                //If the author didn't opt in to the new behavior by using "files", do things the old way
+                return Process(environment, actionConfig, templateCreationResult, outputBasePath);
+            }
+
+            if (targetFiles is null)
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.CouldntDetermineFilesToRestore));
+                return false;
+            }
+
+            foreach (string targetRelativePath in targetFiles)
+            {
+                string pathToRestore = !string.IsNullOrEmpty(outputBasePath) ? Path.GetFullPath(Path.Combine(outputBasePath, targetRelativePath)) : targetRelativePath;
+
+                if (string.IsNullOrEmpty(pathToRestore) ||
+                    (!Directory.Exists(pathToRestore)
+                        && !Path.GetExtension(pathToRestore).EndsWith("proj", StringComparison.OrdinalIgnoreCase)
+                        && !Path.GetExtension(pathToRestore).Equals(".sln", StringComparison.OrdinalIgnoreCase)
+                    ))
+                {
+                    continue;
+                }
+
+                Reporter.Output.WriteLine(string.Format(LocalizableStrings.RunningDotnetRestoreOn, pathToRestore));
+
+                // Prefer to restore the project in-proc vs. creating a new process.
+                bool succeeded = Callbacks.RestoreProject == null ?
```

Comment:
Do I understand it correctly that we don't want to have a similar callback for Build because starting a new dotnet process is by design here? We want to start it, return to the command line, and let it finish "in the background", is that correct?

Reply:
https://github.com/ladipro/one-pagers/pull/18/files is what I'm looking at.

My first goal is to get it to build after running restore. Afterwards, I'm planning to consider what happens if something goes wrong (as with the user calling it explicitly) or just replace it with a conglomeration of some of the other ideas here.

I am intentionally starting up a new process, yes.